### PR TITLE
Cpid diet

### PR DESCRIPTION
**Pull request by denravonska**
Removed unused variables from CPID structures and changed the variable types from double to integers where they are supposed to be integers. This gives us a tiny size decrease of the StructCPID struct. I also removed the casts to double where the variables are automatically promoted to doubles anyway. This doesn't change anything but makes the code cleaner.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -6210,7 +6209,7 @@ StructCPID GetInitializedStructCPID2(std::string name, std::map<std::string, Str
 		{
 				cpid = GetStructCPID();
 				cpid.initialized=true;
-				cpid.LowLockTime = 99999999999;
+				cpid.LowLockTime = std::numeric_limits<unsigned int>::max();
```
Comment (denravonska) [inline src/main.cpp:6212]: Will this work or do we need to use that exact magic number?

File: src/main.cpp [hunk 2]
```diff
@@ -6210,7 +6209,7 @@ StructCPID GetInitializedStructCPID2(std::string name, std::map<std::string, Str
 		{
 				cpid = GetStructCPID();
 				cpid.initialized=true;
-				cpid.LowLockTime = 99999999999;
+				cpid.LowLockTime = std::numeric_limits<unsigned int>::max();
 				cpid.HighLockTime = 0;
 				cpid.LastPaymentTime = 0;
 				cpid.EarliestPaymentTime = 99999999999;
```
Comment (denravonska) [inline src/main.cpp:6215]: See LowLockTime, also change it to match?

**Comment by tomasbrod** [comment]
Good you removed so many unused fields and doubles. I would be careful with the `LowLockTime`, it isn't clear what it means. I browsed the code and saw that `LowLockTime` is used only in debug info to `entry.push_back(Pair("Earliest Payment Time",TimestampToHRDate(stCPID.LowLockTime)));`. Isn't it the same as `EarliestPaymentTime`?

**Comment by denravonska** [comment]
@tomasbrod It looks like it just needs to be a large value so it's possible to do `if(time < LowLockTime)` later on. The reason I changed it was because the previous magic value was too large to fit in a 32bit integer, and all we do is assign 32bit values to it (which is why I changed its signature).

**Comment by grctest** [comment]
Hey guys, just a quick question - are you running these changes in testnet or on production? Because we've got issues with 'bad cpids' causing users to lose sync with the network: https://github.com/gridcoin/Gridcoin-Research/issues/246

**Comment by denravonska** [comment]
@grctest I think I might be running the dev branch in production on one client. I have to check that. These "dangerous" commits which are still in PRs aren't running at all on my end.

Perhaps it could be an idea to force the dev branch to run on testnet unless explicitly stated via a command line argument? Instead of passing it `-testnet` you'd have to pass it `-production` when running the dev branch.

**Comment by philipswift** [comment]
For more detail in the logs add this command in the conf file
debug=true 
source: http://wiki.gridcoin.us/Gridcoinresearch_config_file
I am doing some tests to try and track down the source of the entries in the logs. It mentions duplicate CPIDS and database errors in my logs.

Off topic is this error, the wallet used to do a cheesy hello....
ar 24 2017 12:15, Error while Speaking Good Afternoon  phil.swift, Your Magnitude is 0Conversion from string ",in Production." to type 'Double' is not valid.

Then I also found this 
{
"Command" : "beaconstatus"
},
{
"CPID" : "cca8548e6f66693686316f593c8702e2",
"Beacon Exists" : "No",
"Beacon Timestamp" : "1-1-1970 00:00:00",
"Public Key" : "",
"Private Key" : "308201130201010420d692e06c3ae31d12ef19138ac1574a6e90cd5af9ccd6c89e6e64107d56f3c192a081a53081a2020101302c06072a8648ce3d0101022100fffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2f300604010004010704410479be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798483ada7726a3c4655da4fbfc0e1108a8fd17b448a68554199c47d08ffb10d4b8022100fffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141020101a14403420004322ad753a15c42d76b0eda4c75f1419cb431355bfba1db5881f49718c6b6d6663d7049b19d9255176b8e429addc06674b4463137afc76fa1d26d4cd94c71ac59",
"Local Configuration Public Key" : "04322ad753a15c42d76b0eda4c75f1419cb431355bfba1db5881f49718c6b6d6663d7049b19d9255176b8e429addc06674b4463137afc76fa1d26d4cd94c71ac59",
"Magnitude (As of last superblock)" : 0.00000000,
"Warning" : "Your magnitude is 0 as of the last superblock: this may keep you from staking POR blocks.",
"Errors" : "Public Key Missing. ",
"Help" : "Note: If your beacon is missing its public key, or is not in the chain, you may try: execute advertisebeacon.",
"Configuration Status" : "FAIL"
}
This sniffs of node clock times being out or server times being out. It mentions 1970 above at 'Beacon Timestamp'. Sometimes this id due to a low voltage CMOS battery. Can I suggest anyone with a board using a CMOS battery replaces it and resets the time. The time and date stamps are really critical to syncing up data.

**Pull request merged**